javalib/src/main/scala/java/util/concurrent/locks/ReentrantLock.scala

```diff
@@ -0,0 +1,48 @@
+package java.util.concurrent.locks
+
+import java.io.Serializable
+import java.lang.Thread
+import java.util.concurrent.TimeUnit
+
+class ReentrantLock(fair: Boolean) extends Lock with Serializable {
+
+  private var locked = 0
+
+  def this() = this(false)
+
+  def lock(): Unit = locked += 1
+
+  def lockInterruptibly(): Unit =
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
+    else locked += 1
+
+  def tryLock(): Boolean = {
+    locked += 1
+    true
+  }
+
+  def tryLock(time: Long, unit: TimeUnit): Boolean = {
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
+    tryLock()
```

Comment:
Be consistent over how you decide to reuse or not existing methods. In `lockInterruptibly()` you do `locked += 1` (instead of calling `lock()`), but here you delegate to `tryLock()`. Pick one and stick to it.